superdesk/commands/reassign_metadata.py

```diff
@@ -0,0 +1,56 @@
+import superdesk
+
+
+class ReassignMetadataCommand(superdesk.Command):
+    '''
+
+    Update metadata for a document in both Mongodb and ElasticSearch by supplying a GUID
+
+    manage.py app:update:metadata
+        -i urn:newsml:localhost:2016-09-12T12:11:40.160498:7237e59f-c42d-4865-aee5-e364aeb2966a
+        -m anpa_category
+        -v x
+
+    -m source -v AAP (value is a string name value)
+    -m anpa_category -v x (value is a qcode from vocabularies.json id=categories)
+
+    '''
+    option_list = [
+        superdesk.Option('--guids', '-i', dest='guids', required=True,
+                         help='pass guids of articles to update a metadata item'),
+        superdesk.Option('--meta', '-m', dest='key', required=True),
+        superdesk.Option('--value', '-v', dest='value', required=True)
+    ]
+
+    def run(self, guids, key, value):
+        guids = guids.split(",")
+        category_list = superdesk.get_resource_service('vocabularies').find_one(req=None, _id='categories')
+        category_dict = {entry['qcode']: entry for entry in category_list['items']}
+        patched = value
+
+        try:
+            if key not in ['anpa_category', 'source']:
+                raise Exception('Invalid meta value to update.')
+
+            # http://blog.labix.org/2008/06/27/watch-out-for-listdictkeys-in-python-3
+            if key == 'anpa_category' and value not in list(category_dict.keys()):
+                raise Exception('Setting an invalid category metadata qcode.')
+
+            if key == 'anpa_category':
+                entry = category_dict[value]
+                patched = [{
+                    'qcode': entry.get('qcode'),
+                    'name': entry.get('name'),
+                    'subject': entry.get('subject', None),
+                    'scheme': entry.get('scheme', None)
+                }]
+
+            service = superdesk.get_resource_service('archive')
+
+            for guid in guids:
+                print(service.patch(guid, {key: patched}))
+        except Exception as e:
+            print(e)
```

Comment:
You need there:
```
raise SystemExit(e)
```
Instead of just `print` because command fails and you want to see code 1 instead of 0.

ref: https://docs.python.org/3.5/library/exceptions.html#SystemExit